enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/SlottedPipeBuilder.scala

```diff
@@ -160,9 +160,9 @@ class SlottedPipeBuilder(fallback: PipeBuilder,
 
       case Optional(inner, symbols) =>
         val nullableKeys = inner.availableSymbols -- symbols
-        val nullableOffsets = nullableKeys.map(k => slots.getLongOffsetFor(k.name))
+        val nullableSlots = nullableKeys.map(k => slots.get(k.name).get)
         val argumentSize = physicalPlan.argumentSizes(plan.assignedId)
-        OptionalSlottedPipe(source, nullableOffsets.toSeq, slots, argumentSize)(id)
+        OptionalSlottedPipe(source, nullableSlots.toSeq, slots, argumentSize)(id)
```

Comment:
`nullableSlots.toSeq` will create a `List` which is surprisingly slow in almost all cases, I think `nullableSlots.toIndexedSeq` would be better or using an array would be even better.